board.cpp

```diff
@@ -0,0 +1,118 @@
+#include <fstream>
+#include <random>
+#include <iostream>
+#include "Board.h"
+
+std::default_random_engine generator;
+std::uniform_int_distribution<int> distribution(1, 6);
+
+Board::Board(std::string inputFilePath, std::string outputFilePath)
+{
+    in_path.open(inputFilePath);
+    out_path.open(outputFilePath);
+    if(!in_path.is_open() || !out_path.is_open())
+    {
+        std::cerr << "Error in opening input or output file" << std::endl;
+        exit(-1);
+    }
+    initiateBoardGame();
+}
+
+void Board::initiateBoardGame()
+{
+    int numSnakes, numLadders, numPlayers;
+    in_path >> this->size;
+    in_path >> numSnakes;
+    for (int i = 0; i < numSnakes; i++)
+    {
+        int low, high;
+        in_path >> high >> low;
+        Snake_Ladder *s = new Snake(low, high);
+        this->components.push_back(s);
+    }
+    in_path >> numLadders;
+    for (int i = 0; i < numLadders; i++)
+    {
+        int low, high;
+        in_path >> low >> high;
+        Snake_Ladder *s = new Ladder(low, high);
+        this->components.push_back(s);
+    }
+    in_path >> numPlayers;
+    for (int i = 0; i < numPlayers; i++)
+    {
+        std::string name;
+        in_path >> name;
+        GamePlayer *p = new GamePlayer(name);
+        this->players.push_back(p);
+    }
+}
+
+void Board::startGame()
+{
+
+    out_path << "-------------------------------------------------------" << std::endl;
+    out_path << "-------------------------------------------------------" << std::endl;
+    out_path << "-------------------------------------------------------" << std::endl;
+    int activePlayers = this->players.size();
+    while (activePlayers > 1)
+    {
+        for (int i = 0; i < this->players.size(); i++)
+        {
+            if (!this->players[i]->getActiveStatus())
+                continue;
+
+            int k = this->players[i]->rollDice();
+            int pos_init = this->players[i]->getCurrPosition();
+            for (Snake_Ladder *comp : this->components)
+            {
+                int modified_pos = comp->modifyPosition(pos_init + k);
```

Comment:
this method does not modify the position of the component. Should be named getModifiedPosition.

Reply:
Also, modified_pos does not make much sense. It is basically new position and not modified position

Reply:
Also, avoid short forms where possible. It reduces readability

Reply:
Makes sense, thanks!